node/network/collator-protocol/src/collator_side/mod.rs

```diff
@@ -372,48 +385,87 @@ async fn distribute_collation<Context>(
 	pov: PoV,
 	result_sender: Option<oneshot::Sender<CollationSecondedSignal>>,
 ) -> Result<()> {
-	let relay_parent = receipt.descriptor.relay_parent;
+	let candidate_relay_parent = receipt.descriptor.relay_parent;
+	let candidate_hash = receipt.hash();
 
-	// This collation is not in the active-leaves set.
-	if !state.view.contains(&relay_parent) {
-		gum::warn!(
+	let active_leaf = match state.view.get_leaf_by_ancestry(&candidate_relay_parent, Some(id)) {
+		Some(hash) => hash,
+		None => {
+			gum::debug!(
+				target: LOG_TARGET,
+				para_id = %id,
+				candidate_relay_parent = %candidate_relay_parent,
+				candidate_hash = ?candidate_hash,
+				"Candidate relay parent is not part of the implicit view",
+			);
+			return Ok(())
+		},
+	};
+
+	// TODO [now]: Figure out sanity checks.
```

Comment:
Proposed checks are:

1. Candidate's relay parent is allowed in the implicit view
2. There exists hypothetical membership in the **leaf's** fragment tree
3. No group rotations happened in-between `candidate_relay_parent + 1` and `leaf` inclusive. Otherwise, it means that our group is already working on a different core.